exporter/awsprometheusremotewriteexporter: signingRoundTripper leaks and does not close http.Request body

**Describe the bug**
Noticed while auditing the aws remote write exporter to start fixing the out of order bug, and I saw this code
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d50d10a41e57f946777128daf9c4591ce342c14a/exporter/awsprometheusremotewriteexporter/auth.go#L42-L71

No where is there a reqBody.Close

**Steps to reproduce**
. This test should exhibits the problem
```go
func TestLeakingBody(t *testing.T) {
        awsCreds := fetchMockCredentials()
        authConfig := AuthConfig{Region: "region", Service: "service"}

        server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
                _, err := v4.GetSignedRequestSignature(r)
                assert.NoError(t, err)
                w.WriteHeader(200)
        }))
        defer server.Close()

        serverURL, err := url.Parse(server.URL)
        assert.NoError(t, err)

        setting := confighttp.HTTPClientSettings{
                Endpoint:        serverURL.String(),
                TLSSetting:      configtls.TLSClientSetting{},
                ReadBufferSize:  0,
                WriteBufferSize: 0,
                Timeout:         0,
                CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) {
                        return createSigningRoundTripperWithCredentials(authConfig, awsCreds, next)
                },
        }
        client, _ := setting.ToClient()
        checker := &checkCloser{Reader: strings.NewReader("a=1&b=2")}
        req, err := http.NewRequest("POST", setting.Endpoint, checker)
        assert.NoError(t, err)
        req.GetBody = func() (io.ReadCloser, error) {
                checker.Reader = strings.NewReader("a=1&b=2")
                return checker, nil
        }
        _, err = client.Do(req)
        assert.NoError(t, err)
        assert.True(t, checker.closed)
}
```

which produces
```shell
$ go test -run="Leak" -v
=== RUN   TestLeakingBody
    auth_test.go:116: 
        	Error Trace:	auth_test.go:116
        	Error:      	Should be true
        	Test:       	TestLeakingBody
--- FAIL: TestLeakingBody (0.00s)
FAIL
exit status 1
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsprometheusremotewriteexporter	0.561s
```

**What did you expect to see?**
That test should succeed.

**What did you see instead?**
```shell
=== RUN   TestLeakingBody
    auth_test.go:116: 
        	Error Trace:	auth_test.go:116
        	Error:      	Should be true
        	Test:       	TestLeakingBody
--- FAIL: TestLeakingBody (0.00s)
```

**What version did you use?**
Latest code at 82bc6c3890ae8007f03f34f3c78e5984458338c1

**What config did you use?**
Irrelevant

**Environment**
Irrelevant

/cc @rakyll @alolita @Aneurysm9

--- a/exporter/awsprometheusremotewriteexporter/auth.go
+++ b/exporter/awsprometheusremotewriteexporter/auth.go
@@ -47,6 +47,7 @@ func (si *signingRoundTripper) RoundTrip(req *http.Request) (*http.Response, err
 
 	// Get the body
 	content, err := ioutil.ReadAll(reqBody)
+	reqBody.Close()
 	if err != nil {
 		return nil, err
 	}

--- a/exporter/awsprometheusremotewriteexporter/auth_test.go
+++ b/exporter/awsprometheusremotewriteexporter/auth_test.go
@@ -17,10 +17,12 @@ package awsprometheusremotewriteexporter
 
 import (
 	"errors"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
 	"strings"
+	"sync"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws/credentials"
@@ -61,6 +63,57 @@ func TestRequestSignature(t *testing.T) {
 	assert.NoError(t, err)
 	_, err = client.Do(req)
 	assert.NoError(t, err)
+}
+
+type checkCloser struct {
+	io.Reader
+	mu     sync.Mutex
+	closed bool
+}
+
+func (cc *checkCloser) Close() error {
+	cc.mu.Lock()
+	cc.closed = true
+	cc.mu.Unlock()
+	return nil
+}
+
+func TestLeakingBody(t *testing.T) {
+	// Some form of AWS credentials must be set up for tests to succeed
+	awsCreds := fetchMockCredentials()
+	authConfig := AuthConfig{Region: "region", Service: "service"}
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := v4.GetSignedRequestSignature(r)
+		assert.NoError(t, err)
+		w.WriteHeader(200)
+	}))
+	defer server.Close()
+
+	serverURL, err := url.Parse(server.URL)
+	assert.NoError(t, err)
+
+	setting := confighttp.HTTPClientSettings{
+		Endpoint:        serverURL.String(),
+		TLSSetting:      configtls.TLSClientSetting{},
+		ReadBufferSize:  0,
+		WriteBufferSize: 0,
+		Timeout:         0,
+		CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) {
+			return createSigningRoundTripperWithCredentials(authConfig, awsCreds, next)
+		},
+	}
+	client, _ := setting.ToClient()
+	checker := &checkCloser{Reader: strings.NewReader("a=1&b=2")}
+	req, err := http.NewRequest("POST", setting.Endpoint, checker)
+	assert.NoError(t, err)
+	req.GetBody = func() (io.ReadCloser, error) {
+		checker.Reader = strings.NewReader("a=1&b=2")
+		return checker, nil
+	}
+	_, err = client.Do(req)
+	assert.NoError(t, err)
+	assert.True(t, checker.closed)
 }
 
 func TestGetCredsFromConfig(t *testing.T) {

